spec/workers/nullify_subscribers_worker_spec.rb

```diff
@@ -0,0 +1,35 @@
+RSpec.describe NullifySubscribersWorker do
+  describe ".perform" do
+    let(:nullifyable_time) { 29.days.ago }
+
+    it "nullifies subscribers that don't have any recent active subscriptions" do
+      subscriber = create(:subscriber, created_at: nullifyable_time)
+      create(:subscription, :ended, ended_at: nullifyable_time, subscriber: subscriber)
+
+      expect { subject.perform }
+        .to change { Subscriber.nullified.count }.by(1)
+    end
+
+    it "nullifies subscribers created after the nullifyable period that don't have any subscriptions" do
+      create(:subscriber, created_at: nullifyable_time)
+
+      expect { subject.perform }
+        .to change { Subscriber.nullified.count }.by(1)
+    end
+
+    it "doesn't nullify subscribers with recently ended subscriptions" do
+      subscriber = create(:subscriber, created_at: nullifyable_time)
```

Comment:
does setting nullifyable_time impact this test at all?

Reply:
Yep it does otherwise the test would pass without actually testing the logic I wanted it to, see [here](https://github.com/alphagov/email-alert-api/pull/1462/files#diff-8ddc5cf4620c59f09c23573d63d8540c885cf3f6557988e8935dbd31ebaa0f01R18). I have rejigged the spec to more explicit which should clear things up.